data/campaigns/Descent_Into_Darkness/scenarios/12_Endless_Night.cfg

```diff
@@ -45,50 +50,89 @@
     [side]
         side=2
         controller=ai
-        gold=0
-        # Place leader, grant gold and recruits later
+        # Place leader, grant gold, income and recruits later
         no_leader=yes
-
-        # After I have greatly reduced the number of villages in this scenario
-        # it might be a good idea to give our foolish hero some income
-        {INCOME 5 7 9}
-
-        [ai]
-            villages_per_scout=10
-            village_value=0.5
-        [/ai]
+        gold=0
+        {NO_INCOME}
+        # Set ai values later
     [/side]
 
     {STARTING_VILLAGES 1 6}
 
     [event]
         name=prestart
 
+        {VARIABLE_OP timesForever add 1}
+
         [objectives]
             side=1
+            delayed_variable_substitution=yes #so timesForever note works correctly when using debug
             [objective]
                 condition=win
                 description= _ "Defeat the foolish hero"
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever less_than_equal_to 1}
+                [/show_if]
             [/objective]
+
+            [objective]
+                condition=win
+                description= _ "Defeat another foolish hero"
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever numerical_equals 2}
+                [/show_if]
+            [/objective]
+
+            [objective]
+                condition=win
+                description= _ "Defeat yet another foolish hero"
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever greater_than_equal_to 3}
+                [/show_if]
+            [/objective]
+
             [objective]
                 condition=lose
                 description= _ "Destruction of Mal Keshar"
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever less_than_equal_to 3}
+                [/show_if]
+            [/objective]
+
+            [objective]
+                {ALTERNATIVE_OBJECTIVE_CAPTION}
+                condition=win
+                description= _ "Destruction of Mal Keshar"
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever greater_than_equal_to 4}
+                [/show_if]
             [/objective]
 
             [gold_carryover]
                 carryover_percentage=40
+                [show_if]
+                    {VARIABLE_CONDITIONAL timesForever less_than 10}
+                [/show_if]
             [/gold_carryover]
 
+            [note]
+                description="<b>" + _"Forever" + " " + "$timesForever" + "</b>"
```

Comment:
What's the purpose of this? You're going to have a note that says "Forever 2", "Forever 3", etc.

Reply:
I put it in there so the player will easily know what iteration of the scenario they are on. Also the fact that the text of the counter is "Forever" is another clue to the player that the scenario is repeating deliberately, and I feel it goes well with the intent of the scenario.

Reply:
How can a player who enters the scenario for the first time have any idea what a note that says "Forever 0" means? If they win, then on the next repetition they see "Forever 1", and won't really be any wiser. It'll most likely look more like a bug or like someone accidentally left in some kind of debug counter.

Reply:
It does start at Forever 1 the first time, and would be at Forever 2 on the next repetition.

And the player probably wouldn't have any idea the first time they see it. I figured they would slowly come to the sinking realization of what it means, and that would happen by working with the rest of the changes in that commit (objectives & dialog) to creating a particular atmosphere/feel for the scenario.

I had considered using Endless Night 1, Endless Night 2... but that didn't seem flow as good to me, but maybe that is the way to go?